src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlBulkCopyTest/NoBoxingValuesTypes.cs

```diff
@@ -0,0 +1,487 @@
+
+        private static readonly string _connString = DataTestUtility.TCPConnectionString;
+        private bool _disposedValue;
+
+        private class ItemToCopy
+        {
+            // keeping this data static so the performance of the benchmark is not varied by the data size & shape
+            public int IntColumn { get; } = 123456;
+            public bool BoolColumn { get; } = true;
+        }
+
+        static NoBoxingValueTypes()
+        {
+            _item = new ItemToCopy();
+
+            _items = Enumerable.Range(0, _count).Select(x => _item).ToArray();
+
+            // It would've been great to use mgravell/FastMember here to make thing logic much cleaner and not have to include the custom reader
+            // however, that package is not available on the dotnet-public Nuget source, which is a bummer.
+            _reader = new EnumerableDataReaderFactoryBuilder<ItemToCopy>(_table)
+                .Add("IntColumn", i => i.IntColumn)
+                .Add("BoolColumn", i => i.BoolColumn)
+                .BuildFactory()
+                .CreateReader(_items)
+            ;
+        }
+
+        public NoBoxingValueTypes()
+        {
+            using (var conn = new SqlConnection(_connString))
+            using (var cmd = conn.CreateCommand())
+            {
+                conn.Open();
+                Helpers.TryExecute(cmd, $@"
+                    CREATE TABLE {_table} (
+                        IntColumn INT NOT NULL,
+                        BoolColumn BIT NOT NULL
+                    )
+                ");
+            }
+        }
+
+        private class RunOnceConfig : ManualConfig
+        {
+            public RunOnceConfig()
+            {
+                Add(Job.InProcess.WithLaunchCount(1).WithIterationCount(1).WithWarmupCount(0));
+                Add(MemoryDiagnoser.Default);
+
+                Add(JitOptimizationsValidator.DontFailOnError);
+            }
+        }
+
+
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsNotAzureServer))]
+        public void Should_Not_Box()
+        { // in debug mode, the double boxing DOES occur as the JIT optimizes less code, which causes the test to fail
+#if DEBUG
+            return;
+#else
+            //cannot figure out an easy way to get this to work on all platforms
+
+            var config = new RunOnceConfig(); // cannot use fluent syntax to still support net461
+
+            var summary = BenchmarkRunner.Run<NoBoxingValueTypesBenchmark>(config);
+
+            var numValueTypeColumns = 2;
+            var totalBytesWhenBoxed = IntPtr.Size * _count * numValueTypeColumns;
+
+            var report = summary.Reports.First();
+
+            Assert.Equal(1, report.AllMeasurements.Count);
+            Assert.True(report.GcStats.BytesAllocatedPerOperation < totalBytesWhenBoxed);
+#endif
+        }
+
+        public class NoBoxingValueTypesBenchmark
+        {
+            [Benchmark]
```

Comment:
Hi @cmeyertons 

I highly appreciate your contribution here, but please hold on to committing Performance test and benchmarking code to Manual Tests project in this PR. We're going to introduce a new project specifically for Perf testing, that's where we will add benchmark tests.
_(Till then please upload this test as a file attachment, we'll include it to the new perf project)_

/cc @DavoudEshtehari

Reply:
@cheenamalhotra  - okay sounds good no problem -- i'm going to momentarily leave it in to test that the benchmark is successful in `netfx` if that's alright with you.

Reply:
[NoBoxingValuesTypes.zip](https://github.com/dotnet/SqlClient/files/6384644/NoBoxingValuesTypes.zip)

@cheenamalhotra ^ attached a zip of the file (uncommented out)